src/test/java/api/support/http/InterfaceUrls.java

```diff
@@ -92,10 +92,6 @@ static URL usersUrl(String subPath) {
     return APITestContext.viaOkapiModuleUrl("/users" + subPath);
   }
 
-  static URL calendarUrl(String subPath) {
-    return APITestContext.viaOkapiModuleUrl("/calendar/periods"+ subPath);
-  }
```

Comment:
Why has this been removed, considering that FakeOkapi has been extended to handle this endpoint?

Reply:
It hasn't been used anywhere, no ResourceClient has been created with it. If I understand the logic, typically these are used for fixtures and we don't have OpeningHours fixture. FakeOkapi has been extended to handle /calendar/periods because it is used by OverduePeriodCalculatorService (CIRC-548). We agreed that I will implement API tests for it in CIRC-524, so now it's called by "check in by barcode" API test.

Reply:
> It hasn't been used anywhere, no ResourceClient has been created with it. If I understand the logic, typically these are used for fixtures and we don't have OpeningHours fixture. 

Ok.

> FakeOkapi has been extended to handle /calendar/periods because it is used by OverduePeriodCalculatorService (CIRC-548). 

Please can this be implemented like the other fake APIs, where the relevant test records are set up via the API itself, rather than hard coded values being returned.

This makes it more likely that the API tests will once again be useable with the real API implementations.

Reply:
I think it wouldn't be easy to set up calendar test records and get them back without repeating a big chunk of calendar module logic in the circulation tests. I mean, calendar periods are created with this endpoint https://s3.amazonaws.com/foliodocs/api/mod-calendar/calendar.html#calendar_periods__servicepointid__period_post and then we use this endpoint to get the info about opening hours https://s3.amazonaws.com/foliodocs/api/mod-calendar/calendar.html#calendar_periods_get It returns some kind of compilation, a schedule based on multiple periods. It doesn't even have any IDs. To me it looks like we'd need to repeat the same logic in the tests if we don't want to hard code this response.
Also, considering that this story is blocking a bunch of FE stories, probably it makes sense to create another "technical debt" story to avoid hard coded records in all calendar API tests? It seems like /calendar/periods/:id/period currently is returning hard coded values too.

Reply:
> I think it wouldn't be easy to set up calendar test records and get them back without repeating a big chunk of calendar module logic in the circulation tests. I mean, calendar periods are created with this endpoint https://s3.amazonaws.com/foliodocs/api/mod-calendar/calendar.html#calendar_periods__servicepointid__period_post and then we use this endpoint to get the info about opening hours https://s3.amazonaws.com/foliodocs/api/mod-calendar/calendar.html#calendar_periods_get It returns some kind of compilation, a schedule based on multiple periods. It doesn't even have any IDs. To me it looks like we'd need to repeat the same logic in the tests if we don't want to hard code this response.

Alas, I thought that might be your answer. I think the coupling between these modules is unfortunate due to this API design.

> Also, considering that this story is blocking a bunch of FE stories, probably it makes sense to create another "technical debt" story to get rid of hard coded records in all calendar-related API tests? It seems like /calendar/periods/:id/period currently is returning hard coded values too.

Please feel free to defer this to a separate technical debt issue. Where possible, please organise the calendar related fakes into one place, at least then it is easier to understand the extent of the challenge.